src/main/java/org/jenkinsci/plugins/durabletask/ProcessLiveness.java

```diff
@@ -89,33 +95,26 @@ private static boolean _isAlive(VirtualChannel channel, int pid, Launcher launch
     }
 
     private static final class Liveness extends MasterToSlaveCallable<Boolean,RuntimeException> {
+        private static final POSIX jnr = PosixAPI.jnr();
         private final int pid;
         Liveness(int pid) {
             this.pid = pid;
         }
         @Override public Boolean call() throws RuntimeException {
-            // JNR-POSIX does not seem to work on FreeBSD at least, so using JNA instead.
-            LibC libc = LibC.INSTANCE;
-            if (libc.getpgid(0) == -1) {
-                throw new IllegalStateException("getpgid does not seem to work on this platform");
+            try { // loadPOSIX sends problems to stderr, useless if we are on a slave
+                Method loadNativePOSIX = POSIXFactory.class.getDeclaredMethod("loadNativePOSIX", POSIXHandler.class);
+                loadNativePOSIX.setAccessible(true);
+                loadNativePOSIX.invoke(null, new DefaultPOSIXHandler());
+            } catch (NoSuchMethodException x) {
+                LOGGER.log(Level.WARNING, "loadNativePOSIX no longer exists?", x);
```

Comment:
It worth trying LibC in this case. If you wrap it by `isLinux()` check and proper logging in `else`, it should be safe enough